pkg/controllers/user/endpoints/endpoints.go

```diff
@@ -208,7 +228,7 @@ func convertHostPortToEndpoint(pod *corev1.Pod, clusterName string, node *manage
 }
 
 func publicEndpointToString(p v3.PublicEndpoint) string {
-	return fmt.Sprintf("%s_%s_%v_%s_%s_%s_%s_%s", p.NodeName, p.Address, p.Port, p.Protocol, p.ServiceName, p.PodName, p.Hostname, p.Path)
+	return fmt.Sprintf("%s_%v_%v_%s_%s_%s_%s_%s_%s", p.NodeName, p.Addresses, p.Port, p.Protocol, p.ServiceName, p.PodName, p.IngressName, p.Hostname, p.Path)
```

Comment:
@orangedeng we should do something special with addresses. The arrays comparison can't be done similar to string name. We should sort, and concat the addresses so the same set appears the same way